### PR TITLE
Add access to the entityManager from a JPA PanacheEntity / Repository #7233

### PR DESCRIPTION
**Pull request by italogustavoborba**
#7233

### CODE REVIEW

**Comment by FroMage** [comment]
LGTM, could you just add javadocs to the methods please?

**Comment by emmanuelbernard** [comment]
should we have a line or two in the Panache Guide?

**Comment by FroMage** [comment]
Not sure it's worth it, I think most people will find it with completion.

**Comment by nimo23** [comment]
Why not putting 

```
EntityManager entityManager() {
    return JpaOperations.getEntityManager();
}
```

within `PanacheEntityBase` (and `PanacheRepositoryBase`) instead of its child classes? With the actual version the `entityManager()` is not available when extending those classes directly because of, for example:

> If you want a custom ID type or strategy, you can directly extend {@link PanacheEntityBase}
> instead, and write your own ID field.

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java [hunk 1]
```diff
@@ -38,6 +41,10 @@
     public String toString() {
         return this.getClass().getSimpleName() + "<" + id + ">";
     }
+    
+    public EntityManager entityManager() {
```
Comment (nimo23) [inline extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java]: Why not put this method within PanacheEntityBase instead?

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheRepository.java [hunk 2]
```diff
@@ -13,4 +17,7 @@
  */
 public interface PanacheRepository<Entity> extends PanacheRepositoryBase<Entity, Long> {
 
+	default EntityManager entityManager() {
```
Comment (nimo23) [inline extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheRepository.java]: Why not put this method within PanacheRepositoryBase instead?

**Comment by emmanuelbernard** [comment]
Ah yes @nimo23 is right, the entityManager method should in on `PanacheEntityBase` and `PanacheRepositoryBase` instead. Can you do that change @italogustavoborba ?

**Comment by emmanuelbernard** [comment]
A small test would be nice too BTW :)

**Comment by loicmathieu** [comment]
@italogustavoborba can you provides the methods inside the base class so we can merge this PR ?

**Comment by gastaldi** [comment]
What's the status of this PR? @italogustavoborba can you perform the requested changes above?

**Comment by gastaldi** [comment]
Superseded by https://github.com/quarkusio/quarkus/pull/10541, closing now

**Pull request closed without merging**